lib/metasploit/framework/compiler/win32.rb

```diff
@@ -0,0 +1,49 @@
+require 'metasm'
+require 'erb'
+require 'metasploit/framework/compiler/utils'
+require 'metasploit/framework/compiler/headers/win32'
+
+module Metasploit
+  module Framework
+    module Compiler
+
+      class Win32
+
+        # Returns the binary of a compiled source.
+        #
+        # @param c_template [String] The C source code to compile.
+        # @param type [Symbol] PE type, either :exe or :dll
+        # @raise [NotImplementedError] If the type is not supported.
+        # @return [String] The compiled code.
+        def self.compile(c_template, type=:exe)
```

Comment:
this looks good. I like that the Metasm bits are abstracted, so we could substitute other compilers as well.

However, I'd note that there are other Windows compilers we may want to support in the future (C#, etc.). Should we call this function 'compile_c' with a mind to the future?